### DISCUSSION TITLE
output of evaluator.run

### DISCUSSION THREAD

**Discussion by vsk-phi** [pytorch/ignite#2943] [General]
Hi,
I want to use the supervised evaluator to collect the prediction results. What i found is that after collecting the state of the evaluator eg `state = evaluator.run(data_loader, max_epochs=1)`, the state.output has y_pred,y only for the last batch. 
Is there a way one can collect the output of all batches, if not necessarily in one go? 
The use case is that, i want to write the output predictions, as well as the file names. I can conceivably do this separately with a simple iteration over the  val_loader with nograd, however i want to know if this can be done in an evaluator run.
Thanks

**Comment by sadra-barikbin** [comment]
Hi @vsk-phi , [`ignite.handlers.EpochOutputStore`](https://pytorch.org/ignite/generated/ignite.handlers.stores.EpochOutputStore.html#ignite.handlers.stores.EpochOutputStore) might be the solution for you. It lets you to access model outputs being collected during an epoch, at the end of the epoch. Those are collected as a list of `(y_pred, y)`s. Does this work for you?

**Comment by vfdev-5** [comment]
Hi @vsk-phi , you can use https://pytorch.org/ignite/generated/ignite.handlers.stores.EpochOutputStore.html#epochoutputstore to collect all outputs.

**Reply by vfdev-5** [reply]
@sadra-barikbin is very quick :)

**Comment by vsk-phi** [comment]
Hi,
Thanks both so much for super fast response. It seems that I can use this; probably I didn’t search the documentation enough ☹
